src/rebar_prv_dialyzer.erl

```diff
@@ -210,20 +212,24 @@ proj_plt_files(State) ->
     PltApps = get_config(State, plt_extra_apps, []) ++ BasePltApps,
     BasePltMods = get_config(State, base_plt_mods, []),
     PltMods = get_config(State, plt_extra_mods, []) ++ BasePltMods,
-    Apps = proj_apps(State),
-    DepApps = proj_deps(State),
-    get_files(State, DepApps ++ PltApps, Apps -- PltApps, PltMods, [], []).
+    DepApps = lists:usort(proj_plt_apps(State) ++ PltApps),
+    get_files(State, DepApps, [], PltMods, [], []).
 
 proj_apps(State) ->
     [ec_cnv:to_atom(rebar_app_info:name(App)) ||
      App <- rebar_state:project_apps(State)].
 
-proj_deps(State) ->
+proj_plt_apps(State) ->
     Apps = rebar_state:project_apps(State),
     DepApps = lists:flatmap(fun rebar_app_info:applications/1, Apps),
+    ProjApps = proj_apps(State),
     case get_config(State, plt_apps, top_level_deps) of
-        top_level_deps -> DepApps;
-        all_deps       -> collect_nested_dependent_apps(DepApps)
+        top_level_deps ->
+            DepApps -- ProjApps;
```

Comment:
That sounds like a breaking change?

Reply:
There seems to be a test for this case and it is passing, would it be possible to explain how to test that change isn't breaking anything? (I can add a test if I know the sequence)